File: D_Wooden_Toy_Festival.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
#include <vector>
#include <climits>

using namespace std;

int solve(vector<int>& arr, int n) {
    if (n <= 3) {
        return 0;
    }

    sort(arr.begin(), arr.end());
    int Min = INT_MAX;

    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            int arrlj = (arr[n - 1] + arr[min(n - 1, j + 1)]) / 2;
            int arrij = (arr[j] + arr[i + 1]) / 2;
            int arrsi = (arr[i] + arr[0]) / 2;

            Min = min(Min, max({abs(arr[n - 1] - arrlj), abs(arr[min(n - 1, j + 1)] - arrlj), abs(arr[j] - arrij), abs(arr[i + 1] - arrij), abs(arr[i] - arrsi), abs(arr[0] - arrsi)}));
        }
    }

    return ceil((double)Min);
}

int main() {
    int t;
    cin >> t;

    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
        vector<int> arr(n);

        for (int j = 0; j < n; j++) {
            cin >> arr[j];
        }

        cout << solve(arr, n) << endl;
    }

    return 0;
}

File: D_Wooden_Toy_Festival.py
from sys import stdin
from math import ceil
input = stdin.readline
def solve(arr,n):
    if n<=3:
        return 0
    arr.sort()
    Min = float("inf")
    i,j = 0,n-1
    while j>=i:
        arrlj = (arr[-1]+arr[min(n-1,j+1)])//2
        arrij = (arr[j]+arr[i+1])//2
        arrsi = (arr[i]+arr[0])//2
        Min = min(Min,max(abs(arr[-1]-arrlj),abs(arr[min(n-1,j+1)]-arrlj),abs(arr[j]-arrij),abs(arr[i+1]-arrij),abs(arr[i]-arrsi),abs(arr[0]-arrsi)))
        if arrsi>arrlj:
            j -= 1
        else:
            i += 1
    return ceil(Min)


for i in range(int(input())):
    n = int(input())
    arr = list(map(int,input().split()))
    print(solve(arr,n))
